src/dsp_tools/commands/ingest_xmlupload/create_resources/upload_xml.py

```diff
@@ -67,15 +69,21 @@ def ingest_xmlupload(
     clients = _get_live_clients(con, config)
     state = UploadState(resources, stash, config, permissions_lookup)
 
-    return execute_upload(clients, state)
+    if not execute_upload(clients, state):
+        success = False
 
+    if not _tidy_up_on_server(con, shortcode, creds.dsp_ingest_url):
+        success = False
```

Comment:
If at a later point of time, someone wants to purge & recreate all resources, but doesn't have the mapping CSV any more. Then it will be very handy if it is still available on the server.

Until now, I only rarely called the clean up route, because I thought "Better keep it than delete it too early". 

But there is an important argument for including it here: If we make it a separate command, no one will ever use it. And then, the servers will become cluttered after a while. So we have to decide which evil is worse:

1. Cluttering the server with old CSVs. If a new ingest for the same project is triggered, new entries would be added to the existing file (?), which would lead to confusion.
2. Deleting the CSV every time, with the danger of permanently losing the CSV if a user accidentally deletes it on their machine.

While writing this, I wonder if in the backend, you should implement an archive for old CSVs? This would contain a history of every bulk-ingest that has ever taken place on that particular server. WDYT, @seakayone @BalduinLandolt @siers ?

Reply:
I would always pick clutter over data loss. But I'm not sure either will be a huge problem in reality, that I find very hard to judge. (How often do we actually do multiple xmluploads for the same project? does it hurt to have all asset mappings in the same file, or is it actually a good thing? How often will it happen that someone wants to purge and recreate a project _and_ didn't save the mapping file? ...)

In the long run we definitely should have a clear trace of everything that happened, including bulk ingests. But I don't think this falls under the scope of this project. So here we just need to make a call on calling cleanup: yes/no.

Reply:
> I would always pick clutter over data loss.

Me too. So I'd prefer not calling `/projects/{shortcode}/bulk-ingest/finalize` at all. Would you be okay with that, @seakayone ? It's you who had the effort of implementing it, so it might be disappointing that it will never be used.